tests/components/climate/test_init.py

```diff
@@ -953,6 +955,65 @@ async def test_no_issue_no_aux_property(
     ) not in caplog.text
 
 
+async def test_humidity_validation(
+    hass: HomeAssistant, register_test_integration: MockConfigEntry
+) -> None:
+    """Test validation for humidity."""
+
+    class MockClimateEntityTemp(MockClimateEntity):
+        """Mock climate class with mocked aux heater."""
+
+        _attr_supported_features = (
+            ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.SWING_MODE
+            | ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.TARGET_HUMIDITY
+        )
+        _attr_target_humidity = 50
+        _attr_target_temperature = 15
+        _attr_target_temperature_high = 18
+        _attr_target_temperature_low = 10
+        _attr_target_temperature_step = PRECISION_WHOLE
+
+        def set_humidity(self, humidity: int) -> None:
+            """Set new target humidity."""
+            self._attr_target_humidity = humidity
+
+    test_climate = MockClimateEntityTemp(
+        name="Test",
+        unique_id="unique_climate_test",
+    )
+
+    setup_test_component_platform(
+        hass, DOMAIN, entities=[test_climate], from_config_entry=True
+    )
+    await hass.config_entries.async_setup(register_test_integration.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.test")
+    assert state.attributes.get(ATTR_HUMIDITY) == 50
+
+    with pytest.raises(
+        ServiceValidationError,
+        match="Provided humidity 1 is not valid. Accepted range is 30 to 99",
+    ) as exc:
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SET_HUMIDITY,
+            {
+                "entity_id": "climate.test",
+                ATTR_HUMIDITY: "1",
```

Comment:
1. Please test that above range is also validated
2. I suggest that you test closer to range
```suggestion
                ATTR_HUMIDITY: "29",
```

Reply:
1 or 29 is both below min so I don't get to put closer to range. If it's off, it's off.